### PR TITLE
Combine OpenPBS & LSF integration tests

### PR DESCRIPTION
**Pull request by pinkwah**
Resolves: #7316 (hopefully)

There is a lot of overlap between the two drivers, so we combine some of the generic integration tests into a single parameterised pytest.

### CODE REVIEW

File: tests/integration_tests/scheduler/conftest.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+@pytest.fixture
+def mock_lsf(pytestconfig, monkeypatch, tmp_path):
+    if pytestconfig.getoption("lsf"):
+        # User provided --lsf, which means we should use the actual LSF
+        # cluster without mocking anything.""
+        return
+    mock_bin(monkeypatch, tmp_path)
+
+
+async def poll(driver: Driver, expected: set[int], *, started=None, finished=None):
```
Comment (berland) [inline tests/integration_tests/scheduler/conftest.py:32]: There is a version of this one in unit_tests/test_lsf_driver.py as well. Maybe that one can also use this common function.

Comment (pinkwah) [reply tests/integration_tests/scheduler/conftest.py:32]: Are you sure? I don't see anything there.

Comment (berland) [reply tests/integration_tests/scheduler/conftest.py:32]: It is well hidden far down... 
https://github.com/equinor/ert/blob/f26a69942191141f1d29998f2fb4010655096502/tests/unit_tests/scheduler/test_lsf_driver.py#L268

Comment (pinkwah) [reply tests/integration_tests/scheduler/conftest.py:32]: Oh, my branch was outdated. Will move `poll` to `tests.utils`.

Comment (berland) [reply tests/integration_tests/scheduler/conftest.py:32]: Maybe also add a docstring explaining its intended usage.

**Review by berland — APPROVED** [review]
💯

**Pull request merged**